vnext/ReactWindowsCore/WinRTWebSocketResource.cpp

```diff
@@ -237,15 +273,36 @@ fire_and_forget WinRTWebSocketResource::PerformWrite()
       self->m_writer.WriteBytes(arr);
     }
 
-    co_await self->m_writer.StoreAsync();
+    auto async = self->m_writer.StoreAsync();
 
-    if (self->m_writeHandler)
+    co_await lessthrow_await_adapter<DataWriterStoreOperation>{async};
+
+    if (SUCCEEDED(async.ErrorCode()))
+    {
+      if (self->m_writeHandler)
+      {
+        self->m_writeHandler(length);
+      }
+    }
+    else
+    {
+      if (self->m_errorHandler)
+      {
+        self->m_errorHandler({ GetRestrictedErrorMessage(), ErrorType::Send });
+      }
+    }
+  }
+  catch (std::exception const& e)
+  {
+    // Utf8ToUtf16 may throw
```

Comment:
>// Utf8ToUtf16 may throw [](start = 4, length = 24)

Then we must use an alternative API that does not throw. In Office we return std::optional and std::nullopt indicates a parse failure. It is better than throwing an exception.

Reply:
Do we have other APIs available in RNW?

Reply:
This is perhaps an enhancement for the UTF API in RNW, if we don't have such non-throwing methods in RNW or C++/WinRT.

Reply:
The Microsoft::Common::Unicode APIs call MultiByteToWideString etc. and it decides to throw or not. So you can just call the Windows API directly without going through this exception based API.

Reply:
This might be a contentious opinion, but in cases like this where the string is trusted, fail-fast seems fine. I.e. call the nice builtin convenience API like `winrt::to_string` that throws and let it hit the noexcept boundary and crash if we have a garbage input.

Downside is extra binary cost from the generated catch + std::terminate, but imo that's small enough to not be worth optimizing for and creating the confusion of our own helpers with special behavior.